artemis-commons/src/main/java/org/apache/activemq/artemis/core/server/ActiveMQScheduledComponent.java

```diff
@@ -251,6 +251,9 @@ public synchronized boolean isStarted() {
    // this will restart the scheduled component upon changes
    private void restartIfNeeded() {
       if (isStarted()) {
+         // it has already been through an initial delay,
+         // now we just use the next interval
+         this.initialDelay = period;
```

Comment:
nice!!!!! :+1:

Reply:
I've built up this test:
```
   @Test
   public void testVerifyInitialDelayChanged() {
      final long initialDelay = 10;
      final long period = 100;
      final ActiveMQScheduledComponent local = new ActiveMQScheduledComponent(scheduledExecutorService, executorService, initialDelay, period, TimeUnit.MILLISECONDS, false) {
         @Override
         public void run() {

         }
      };
      local.start();
      final long newInitialDelay = 1000;
      //the parameters are valid?
      assert initialDelay != newInitialDelay && newInitialDelay != period;
      local.setInitialDelay(newInitialDelay);
      local.stop();
      Assert.assertEquals("the initial dalay can't change", newInitialDelay, local.getInitialDelay());
   }
```
And it fail: maybe is better to create a `start(long forceInitialDelay)` method and call it into the restart, wdyt?

Reply:
Nope. Initial delay should be at the beginnnjng only.  Why would you want a further initial delay when it’s not initial any longer.  Just to keep it simple.

Reply:
@clebertsuconic But if (at the end of the test) I will start the component, it will have a changed configuration...it can lead to subtle bugs...

Reply:
@clebertsuconic it is "initial" on each start after a stop (eg how i use it of JDBC)

Reply:
What do you need an initialDelay different from the frequency anyways? What is the usecase?

Reply:
@clebertsuconic When I have a very big period but I need the first call to happen right after the start of the component. IMO a user expectation is that the component start will honour the configuration provided if the componet was in a stopped state...